visualization_msgs/msg/Marker.msg

```diff
@@ -56,8 +56,13 @@ geometry_msgs/Point[] points
 # NOTE: alpha is not yet used
 std_msgs/ColorRGBA[] colors
 
+# Texture resource is a special URI that can either reference a texture file in
+# a format acceptable to libcurl or:
```

Comment:
We should add to the documentation references to resource retriever and use it like we already do for the meshes. (This lets us use package:// syntax too)

https://index.ros.org/p/resource_retriever/

Then for the embedded approach I might suggest `embedded://name` or `attached://name` and then we can mirror it and used the same keyword for the mesh.